### ISSUE TITLE
A question about cr.pl.gene_trends() of cellrank 

### ISSUE THREAD

**Issue by liuyang218622** [theislab/cellrank#1130]
Hello, 
When I run Cellrank using my scRNA-seq data. I annotated the cell types manually. I want to look into marker gene expression trends using the command below. 
cr.pl.gene_trends(
    adata,
    model=model,
    data_key="MAGIC_imputed_data",
    genes=['Procr','Hlf','Ccl9','Phf14','Itga2b','Fabp5','Ms4a2','Fcer1a', # Myeloid_BMCP
              'Car1','Ces2g', # Myeloid_MEP
              'Elane','Ms4a3','Ltf','Lrg1',  #Myeloid_Granulocytes
              'Flt3','Bex6', #Lymphoid_LMPP,
              'Ccl5','Il2rb', # Lymphoid_NK/T
              'Cd74','Ppfia4', # Lymphoid_DC
              'Cd79a','Ebf1', # Lymphoid_B
              'Mllt3','Cd34', # HSC/MPP
              'Gata1','Klf1','F2r', # MEP
              'Plek', 'Hdc',#BMCP
              'Igll1','Cd19' # B cells/NK/T/cells
          ],
    same_plot=True,
    ncols=2,
    time_key="dpt_pseudotime",
    hide_cells=True,
)
But I got these results. 
![image](https://github.com/theislab/cellrank/assets/13673840/6b4690ae-51f7-49c3-bf65-f56344bba624)
![image](https://github.com/theislab/cellrank/assets/13673840/b03d73e1-d85a-4888-92a5-5cfaa75209a7)
I have a question: 
why are the marker's expressions at the starting point all different?  Since the start point is the HSCs. shouldn't their expression levels be the same in all trajectories? 
Your example pictures in the tutorials look better. 
![image](https://github.com/theislab/cellrank/assets/13673840/9c3f8203-1c55-4583-9f53-944144bc73d9)
Is there something wrong with my cell type annotation? 
...

**Comment by Marius1311** [comment]
Hi @liuyang218622, what's the model in your case?

**Comment by liuyang218622** [comment]
model = cr.models.GAMR(adata)

At 2023-10-17 01:23:43, "Marius Lange" ***@***.***> wrote:

Hi @liuyang218622, what's the model in your case?

**Comment by Marius1311** [comment]
Hi @liuyang218622, I think this is a problem with either model convergence or very low fate probabilities for some trajectories towards their beginning. Please check the model API: https://cellrank.readthedocs.io/en/latest/api/_autosummary/models/cellrank.models.GAMR.html

In particular, the parameters `n_knots` and `smoothing_penalty` might be interesting to you. 

Additionally, please explore the API description for `cr.pl.gene_trends` here: https://cellrank.readthedocs.io/en/latest/api/_autosummary/plotting/cellrank.pl.gene_trends.html

The parameter `weight_threshold` might be interesting, and I would suggest trying something like `weight_threshold=(1e-3, 1e-3)`, as in our feature trend plotting tutorial: https://cellrank.readthedocs.io/en/latest/notebooks/tutorials/estimators/800_gene_trends.html

**Comment by Marius1311** [comment]
Finally, this might also be related to this issue here: https://github.com/theislab/cellrank/issues/1126

**Comment by liuyang218622** [comment]
Thank you for your answer. I tried to adjust the parameters as you suggested. 
For example, I used the command below to do analysis. 
model = cr.models.GAMR(adata,n_knots=6,smoothing_penalty=6)
cr.pl.gene_trends(
    adata,
    model=model,
    data_key="MAGIC_imputed_data",
    genes=['Procr','Hlf','Ccl9','Phf14','Itga2b','Fabp5','Ms4a2','Fcer1a', # Myeloid_BMCP
              'Car1','Ces2g', # Myeloid_MEP
              'Elane','Ms4a3','Ltf','Lrg1',  #Myeloid_Granulocytes
              'Flt3','Bex6', #Lymphoid_LMPP,
              'Ccl5','Il2rb', # Lymphoid_NK/T
              'Cd74','Ppfia4', # Lymphoid_DC
              'Cd79a','Ebf1', # Lymphoid_B
              'Mllt3','Cd34', # HSC/MPP
              'Gata1','Klf1','F2r', # MEP
              'Plek','Hdc',#BMCP
              'Igll1','Cd19' # B cells/NK/T/cells
          ],
    same_plot=True,
    ncols=2,
    time_key="dpt_pseudotime",
    hide_cells=True,
    weight_threshold=(1e-8, 1e-8)
)
But I can get a bit better results, But the start points are still different. 
![image](https://github.com/theislab/cellrank/assets/13673840/adde4105-fb60-45a2-8579-c35a663d9a9a)
![image](https://github.com/theislab/cellrank/assets/13673840/c7e8c866-aba7-44df-a6d4-ebf5978ac8fc)
![image](https://github.com/theislab/cellrank/assets/13673840/11ae0443-f6fc-40a3-996c-c21a744e1094)I saw in your tutorials(https://cellrank.readthedocs.io/en/latest/notebooks/tutorials/general/100_getting_started.html) you said![image](https://github.com/theislab/cellrank/assets/13673840/20e9f84e-ea33-411d-89d3-0827fbe3e7f9) . Is it associated with the HSC cells contribution to different trajectory?

**Comment by liuyang218622** [comment]
of course, after I just the parameters following your suggestions. The results became better than befor. For example: 
The picture below is before: 
![image](https://github.com/theislab/cellrank/assets/13673840/4e42b0e9-e67d-4dfc-a81a-b288de816ef3)
The picture below is after and it looks much better. ![image](https://github.com/theislab/cellrank/assets/13673840/563eaccf-4497-47f1-a2a8-5cda0a1b3b28)

**Comment by Marius1311** [comment]
ok, did you play around with the weight threshold as well? Can you check for fate probabilities of your different trajectories towards the beginning of your pseudotime? I suppose that some trajectories have really low fate probabilities towards the beginning.

**Comment by liuyang218622** [comment]
I used the command below and got these results. 
![image](https://github.com/theislab/cellrank/assets/13673840/c6083822-fd67-46b9-8726-377b7c763d92)
What does the mean that some trajectories have really low fate probabilities towards the beginning? I defined the root as HSC manually. Is it mean maybe the root I set is not the "true" root of some trajectories ?

**Comment by liuyang218622** [comment]
I calculated the first 10, 20 and 30 bins's average value 
![image](https://github.com/theislab/cellrank/assets/13673840/6f813065-cff5-4f2e-886d-24d9fa88160e)
The cell number of granulocytes is highest among the five terminal status.

**Comment by Marius1311** [comment]
ok, it looks like the vast majority of early cells is predicted to become Granulocytes here. Especially in the first bin, average fate probabilities for Dendritic cells are below 1%. That means, there are almost no cells the algorithm can use to reliably draw gene expression trends for the Dendritic cell trajectory. You can resolve this by using the `weight_threshold` I hinted at above. You could start with `weight_threshold=(1e-3, 1e-3)` and increase both number if that's not enough. Essentially, that resolves the ambiguity among early cells to become Dentritic cells and should make early expresion trends (for early pseudotimes) more reliable.

**Comment by liuyang218622** [comment]
Sorry, can I ask you another question? What does the expression on the ordinate of this graph mean? Is it log (normalization expression + 1)? Or is it the result after scaling? 
Because I found in your example pictures of the tutorials(https://cellrank.readthedocs.io/en/latest/notebooks/tutorials/general/100_getting_started.html). We can see the negative value on the ordinate of the graph. 
![image](https://github.com/theislab/cellrank/assets/13673840/0b38296f-fec6-4fdd-b263-0f1865f56b73). 
When I look into the expression trends of MEP marker genes. I found the expression trends of MEP marker genes are shorted   in KO and its expression level is lower in KO. 
![image](https://github.com/theislab/cellrank/assets/13673840/f91699fd-920a-45ca-a808-c447923035a6)
But when I look into the expression levels of these genes in MEP. I get the opposite results. adata.X is the scaled log(normalisation expression+1). The picture below's expression is log2(TPM+1). But I think anyway the tendency should be similar. 
![image](https://github.com/theislab/cellrank/assets/13673840/cfe16300-a5ec-4e72-b871-3303f8861328)

**Comment by liuyang218622** [comment]
After I checked the data. I think the expression on the ordinate of the expression trends plot  is more likely the data from "MAGIC_imputed_data" ? am I right ?

**Comment by Marius1311** [comment]
the y-axis is the GAM-predicted gene expression, smoothed to each trajectory. Input to the GAM is typically imputed data as we assume Gaussian noise by default. In your case, it seems to be MAGIC imputed data.

**Comment by liuyang218622** [comment]
I have another question. 
For example, we found that the MEP marker genes in KO sample have shorter pesudotime and lower expression level 
![image](https://github.com/theislab/cellrank/assets/13673840/240922a7-2238-4240-aae9-d462d233784d)
But when we look into the terminal status of KO and WT. We found that the MEP's terminal status in WT is not the same with KO. Why is it happen? Is this what makes the pesudotime and expression level different? Can I conclude that the MEP marker genes in KO samples displayed shorter pesudotime and lower expression level? After all, the MEP terminal status in KO and WT is different.  
![image](https://github.com/theislab/cellrank/assets/13673840/9ad02bfb-7b74-40c9-85af-31f0c22628ba)
